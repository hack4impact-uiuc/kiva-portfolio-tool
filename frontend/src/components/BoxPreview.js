import React, { Component } from 'react'

import { ContentPreview } from 'box-ui-elements'
import messages from 'box-ui-elements/i18n/en-US'
import 'box-ui-elements/dist/preview.css'
import '../styles/index.scss'

import { getAccessToken } from '../utils/ApiWrapper'

const container = document.querySelector('.container')
const language = 'en-US'
const FILE_ID_VIDEO = '308566420378'
const FILE_ID_3D = '319004423111'
const FILE_ID_TEXT = '308349801521'
const FILE_ID_EXCEL = '319011536090'
const FILE_ID_AUDIO = '308566419514'
const FILE_ID_IMAGE = '308345646235'

class BoxPreview extends Component {
  constructor(props) {
    super(props)
    this.state = {
      fileId: 419075737331,
      accessToken: null
    }
  }

  async componentDidMount() {
    const res = await getAccessToken()
    if (res) {
      this.setState({
        accessToken: res
      })
    } else {
      this.setState({
        accessToken: null
      })
    }
  }

  render() {
    return (
      <ContentPreview
        hasHeader
        fileId={this.state.fileId}
        token={'xNT24dYkz8KM4QBdUnPTLvm3mAxWbgXN'}
        //token={this.state.accessToken}
        language={language}
        messages={messages}
        collection={[
          this.state.fileId,
          419072325627,
            418980052676
        ]}
      />
    )
  }
}

export default BoxPreview
