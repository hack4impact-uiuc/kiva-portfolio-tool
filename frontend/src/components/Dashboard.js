import React, { Component } from 'react'
import { Container, Row, Col, Button } from 'reactstrap'

import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import {
  updateDocuments,
  updateMessages,
  updateInstructions,
  setUserType,
  beginLoading,
  endLoading
} from '../redux/modules/user'

import WithAuth from './auth/WithAuth'
import DocumentList from './DocumentList'
import NavBar from './NavBar'

import {
  getDocumentsByUser,
  getMessagesByFP,
  updateFieldPartnerStatus,
  getFPByID
} from '../utils/ApiWrapper'

import add from '../media/add.png'

const mapStateToProps = state => ({
  isPM: state.user.isPM,
  documents: state.user.documents,
  messages: state.user.messages,
  information: state.user.information
})

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      updateDocuments,
      updateMessages,
      setUserType,
      beginLoading,
      endLoading,
      updateInstructions
    },
    dispatch
  )
}
export class Dashboard extends Component {
  constructor(props) {
    super(props)

    this.state = {
      fp_statuses: ['Missing', 'Rejected', 'Pending', 'Approved'],
      pm_statuses: ['Pending', 'Missing', 'Rejected', 'Approved'],
      pm_id: null
    }

    this.handleFinish = this.handleFinish.bind(this)
  }

  async componentDidMount() {
    /**
     * Contains all documents received from backend
     */

    this.props.beginLoading()
    let documentsReceived = []

    // temporary - REMOVE after auth integration
    documentsReceived = await getDocumentsByUser(this.props.match.params.id)
    this.props.setUserType(this.props.match.params.user === 'pm')

    /**
     * Contains all messages received from backend
     */
    const messagesReceived = await getMessagesByFP(
      this.props.match.params.id,
      this.props.match.params.user === 'fp'
    )

    /**
     * Contains all information received from backend
     */
    const fp = await getFPByID(this.props.match.params.id)
    const instructionsReceived = fp.instructions
    this.setState({ pm_id: fp.pm_id })

    if (documentsReceived) {
      this.props.updateDocuments(documentsReceived)
    } else {
      this.props.updateDocuments([])
    }

    if (messagesReceived) {
      this.props.updateMessages(messagesReceived)
    } else {
      this.props.updateMessages([])
    }

    if (instructionsReceived) {
      this.props.updateInstructions(instructionsReceived)
    } else {
      this.props.updateInstructions('')
    }
    this.props.endLoading()
  }

  /**
   * When a Field Partner has finished the process, this method is called to move their status to 'Complete'
   */
  async handleFinish() {
    this.props.beginLoading()
    await updateFieldPartnerStatus(this.props.match.params.id, 'Complete')
    this.props.history.push('/overview/' + this.state.pm_id)
    this.props.endLoading()
  }

  pStyle = {
    margin: 'auto'
  }

  render() {
    return (
      <div className="background-rectangles maxheight">
        <NavBar inDashboard />
        <Container id="dashboard-container">
          {this.props.isPM ? (
            <Row>
              <Col className="text-centered" md="12">
                <Button
                  className="add-doc-text"
                  color="transparent"
                  onClick={() => this.props.history.push('/setup/' + this.props.match.params.id)}
                >
                  <img className="addImg" src={add} alt="Add icon" />
                  <span className="add-doc-text">Update requirements/instructions</span>
                </Button>
              </Col>
              <Col className="text-centered" md="12">
                <Button color="success" onClick={this.handleFinish}>
                  Finish Process
                </Button>
              </Col>
            </Row>
          ) : null}
          <Row>
            {this.props.documents
              ? this.props.isPM
                ? this.state.pm_statuses.map(key => {
                    return (
                      <Col sm="12" md="6" className="dashboard-width-override">
                        <DocumentList documents={this.props.documents[key]} status={key} />
                      </Col>
                    )
                  })
                : this.state.fp_statuses.map(key => {
                    return (
                      <Col sm="12" md="6" className="dashboard-width-override">
                        <DocumentList documents={this.props.documents[key]} status={key} />
                      </Col>
                    )
                  })
              : null}
          </Row>
        </Container>
      </div>
    )
  }
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WithAuth(Dashboard))
