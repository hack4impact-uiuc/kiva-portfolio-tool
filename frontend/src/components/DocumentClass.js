import React, { Component } from 'react'
import { connect } from 'react-redux'
import DocumentClassPreview from './DocumentClassPreview'
import { Button, Modal, ModalBody, ModalFooter } from 'reactstrap'
import {
  deleteDocumentClass,
  updateDocumentClass,
  getAllDocumentClasses
} from '../utils/ApiWrapper'
import { bindActionCreators } from 'redux'
import { updateDocumentClasses } from '../redux/modules/user'
import { beginLoading, endLoading } from '../redux/modules/auth'
import Dropzone from 'react-dropzone'

import '../styles/documentclasspage.css'

import edit from '../media/greyEdit.png'
import remove from '../media/remove.png'

const mapStateToProps = state => ({})

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      updateDocumentClasses,
      beginLoading,
      endLoading
    },
    dispatch
  )
}

/*
Corresponds to each individual Document Class in the Admin/PM overview page
Name is represented normally
Buttons exist to view, edit, and delete
View shows the description and a Box preview
*/
export class DocumentClass extends Component {
  constructor(props) {
    super(props)

    this.state = {
      name: this.props.documentClass.name,
      description: this.props.documentClass.description,
      editModal: false, // modal that appears when editing a Document class
      deleteModal: false, // modal that appears making sure the user actually wants to delete the document class
      files: []
    }

    this.editToggle = this.editToggle.bind(this)
    this.deleteToggle = this.deleteToggle.bind(this)
    this.handleDelete = this.handleDelete.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  editToggle() {
    this.setState({ editModal: !this.state.editModal })
  }

  deleteToggle() {
    this.setState({ deleteModal: !this.state.deleteModal })
  }

  updateName = event => {
    this.setState({ name: event.target.value })
  }

  updateDescription = event => {
    this.setState({ description: event.target.value })
  }

  async handleSubmit() {
    this.props.beginLoading()
    if (this.state.files.length == 0) {
      await updateDocumentClass(
        this.props.documentClass._id,
        this.state.name,
        this.state.description,
        null,
        null
      )
    } else {
      await updateDocumentClass(
        this.props.documentClass._id,
        this.state.name,
        this.state.description,
        this.state.files[0],
        this.state.files[0].name
      )
    }
    const document_classes = await getAllDocumentClasses()
    if (document_classes) {
      this.props.updateDocumentClasses(document_classes)
    } else {
      this.props.updateDocumentClasses([])
    }
    this.props.endLoading()
    this.editToggle()
  }

  onDrop(files) {
    this.setState({
      files
    })
  }

  async handleDelete() {
    this.props.beginLoading()
    await deleteDocumentClass(this.props.documentClass._id)
    const document_classes = await getAllDocumentClasses()
    if (document_classes) {
      this.props.updateDocumentClasses(document_classes)
    } else {
      this.props.updateDocumentClasses([])
    }
    this.props.endLoading()
    this.deleteToggle()
  }

  render() {
    return (
      <>
        <Modal isOpen={this.state.editModal} toggle={this.editToggle}>
          <ModalBody>
            <form>
              <span> Name: </span>
              <input onChange={this.updateName} value={this.state.name} />
              <br />
              <span> Description: </span>
              <textarea
                name="paragraph_text"
                cols="50"
                rows="10"
                onChange={this.updateDescription}
                value={this.state.description}
              />
              <br />
              <div className="dropPage">
                <section className="droppedBox">
                  <div className="dropZone">
                    <Dropzone onDrop={this.onDrop.bind(this)}>
                      {({ getRootProps, getInputProps }) => (
                        <section>
                          <div {...getRootProps()}>
                            <input {...getInputProps()} />
                            {this.state.files.length > 0 ? (
                              <p>File uploaded: {this.state.files[0].name}</p>
                            ) : (
                              <p>Click to Upload</p>
                            )}
                          </div>
                        </section>
                      )}
                    </Dropzone>
                  </div>
                </section>
                <hr />
              </div>
            </form>
          </ModalBody>
          <ModalFooter>
            <Button className="invalidSearchButton" onClick={this.editToggle}>
              Return
            </Button>
            <Button onClick={this.handleSubmit}>Update Document Class</Button>
          </ModalFooter>
        </Modal>
        <tr className="hoverable">
          {this.props.documentClass.name ? (
            <td data-testid="docClass">{this.props.documentClass.name}</td>
          ) : null}
          <td data-testid="interaction" className="interaction">
            <DocumentClassPreview documentClass={this.props.documentClass} />
            <Button color="transparent" onClick={this.editToggle}>
              <img className="buttonimg" src={edit} />
            </Button>
            <Button color="transparent" onClick={this.deleteToggle}>
              <img className="buttonimg" src={remove} />
            </Button>
            <Modal isOpen={this.state.deleteModal} toggle={this.deleteToggle}>
              <ModalBody>
                <p>
                  Are you sure you want to delete {this.props.documentClass.name}? This will delete
                  every document of this type for all users.
                </p>
              </ModalBody>
              <ModalFooter>
                <Button className="invalidSearchButton" onClick={this.handleDelete}>
                  Delete and return
                </Button>
                <Button className="invalidSearchButton" onClick={this.deleteToggle}>
                  Return
                </Button>
              </ModalFooter>
            </Modal>
          </td>
        </tr>
      </>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DocumentClass)
