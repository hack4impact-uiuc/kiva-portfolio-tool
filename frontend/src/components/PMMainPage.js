import React, { Component } from 'react'
import { getAllPartners, createFieldPartner, getAllPMs } from '../utils/ApiWrapper'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs'
import 'react-tabs/style/react-tabs.css'
import { bindActionCreators } from 'redux'
import { beginLoading, endLoading } from '../redux/modules/auth'
import { connect } from 'react-redux'
import { Progress, Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap'
import '../styles/partnerbar.css'
import search from '../media/search.png'

const mapStateToProps = state => ({
  isPM: state.user.isPM
})

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      beginLoading,
      endLoading
    },
    dispatch
  )
}

export class PMMainPage extends Component {
  constructor(props) {
    super(props)

    this.state = {
      partners: [],
      filtered: [],
      query: '',
      email: '',
      org_name: '',
      modal: false,
      pm_id: null
    }
    this.toggle = this.toggle.bind(this)
    this.handleNewFP = this.handleNewFP.bind(this)
    this.handleClickIP = this.handleClickIP.bind(this)
    this.handleClickNew = this.handleClickNew.bind(this)
  }

  /**
   * Waits for component to load and get all the partners attached to pm
   */
  async componentDidMount() {
    let partners = await getAllPartners()
    this.setState(this.loadPartners(partners))
    let pms = await getAllPMs()
    let pm = pms[0]
    this.setState({ pm_id: pm._id })
  }

  /**
   *
   * @param {*} res updates state of partners and filtered to res if it exists
   */
  loadPartners(res) {
    if (res) {
      return {
        partners: res,
        filtered: res
      }
    } else {
      return {
        partners: [],
        filtered: []
      }
    }
  }

  /**
   * Picks up on any changes in query from and updates query state
   * Filtered is then filter to include substrings of query
   * Then state is updated
   */
  handleQueryChange = event => {
    let newState = this.state
    let query = event.target.value.toLowerCase()
    newState['query'] = query
    newState['filtered'] = []
    if (query == '') {
      newState['filtered'] = this.state.partners
    } else {
      newState['filtered'] = this.state.partners.filter(partner =>
        partner.org_name.toLowerCase().includes(query)
      )
    }
    this.setState(newState)
  }

  handleNameChange = event => {
    this.setState({ org_name: event.target.value })
  }

  handleEmailChange = event => {
    this.setState({ email: event.target.value })
  }

  toggle = () => {
    this.setState({ modal: !this.state.modal })
  }

  async handleNewFP() {
    await createFieldPartner(this.state.org_name, this.state.email, this.state.pm_id)
    let partners = await getAllPartners()
    this.setState(this.loadPartners(partners))
    this.toggle()
  }

  handleClickIP = id => {
    this.props.history.push('/dashboard/pm/' + id)
  }

  handleClickNew = id => {
    this.props.history.push('/selectdocumentspage/' + id)
  }

  render() {
    return (
      <div className="page">
        <Modal isOpen={this.state.modal} toggle={this.toggle}>
          <ModalHeader>Add New Field Partner</ModalHeader>
          <ModalBody>
            <form onSubmit={this.handleNewFP}>
              <p>Organization Name:</p>
              <input
                type="text"
                value={this.state.name}
                size="50"
                placeholder="Enter the Field Partner's organization name here..."
                onChange={this.handleNameChange}
              />
              <p>Email:</p>
              <input
                type="text"
                value={this.state.email}
                size="50"
                placeholder="Enter the Field Partner's email here..."
                onChange={this.handleEmailChange}
              />
            </form>
          </ModalBody>
          <ModalFooter>
            <Button onClick={this.toggle}>Exit</Button>
            <Button onClick={this.handleNewFP}>Create</Button>
          </ModalFooter>
        </Modal>
        <div>
          <span />
          <p>Fieldy McPartnerson</p>
        </div>
        <h2>Field Partners</h2>
        <Button onClick={this.toggle}>Add New Field Partner</Button>
        <form onSubmit={this.handleSubmit}>
          <img src={search} width="25" />
          <input
            className="input-master"
            type="text"
            value={this.state.query}
            placeholder="Search for a Field Partner..."
            onChange={this.handleQueryChange}
          />
        </form>

        <Tabs className="tab-master">
          <TabList className="react-tabs__tab-list">
            <Tab>In Process</Tab>
            <Tab>New Partner</Tab>
            <Tab>Complete</Tab>
          </TabList>

          <TabPanel>
            <div className="partnerPanel">
              {this.state.filtered
                .filter(partner => partner.app_status == 'In Process')
                .map(partner => {
                  return (
                    <Button
                      className="partnerButton"
                      color="transparent"
                      onClick={() => this.handleClickIP(partner._id)}
                    >
                      <PartnerBar partner={partner} />
                    </Button>
                  )
                })}
            </div>
          </TabPanel>

          <TabPanel>
            <div className="partnerPanel">
              {this.state.filtered
                .filter(partner => partner.app_status == 'New Partner')
                .map(partner => {
                  return (
                    <Button
                      className="partnerButton"
                      color="transparent"
                      onClick={() => this.handleClickNew(partner._id)}
                    >
                      <PartnerBar partner={partner} />
                    </Button>
                  )
                })}
            </div>
          </TabPanel>

          <TabPanel>
            <div className="partnerPanel">
              {this.state.filtered
                .filter(partner => partner.app_status == 'Complete')
                .map(partner => {
                  return (
                    <Button
                      className="partnerButton"
                      color="transparent"
                      onClick={() => this.handleClickNew(partner._id)}
                    >
                      <PartnerBar partner={partner} />
                    </Button>
                  )
                })}
            </div>
          </TabPanel>
        </Tabs>
      </div>
    )
  }
}

/**
 * Component containing information about a single partner
 */
class PartnerBar extends Component {
  constructor(props) {
    super(props)
  }

  /**
   * Calculates percentages of each state of a document and puts it into a progress bar
   * In addition prints all info of a field partner
   */
  render() {
    const partner = this.props.partner
    const documents = partner.documents

    let approved = 0
    let pending = 0
    let rejected = 0
    let rest = 0

    // counts number of documents in eachS
    let len = documents.length
    for (const document in documents) {
      let item = documents[document].status
      if (item == 'Approved') {
        approved += 1
      } else if (item == 'Pending') {
        pending += 1
      } else if (item == 'Rejected') {
        rejected += 1
      }
    }

    // turns into percentage values
    if (len > 0) {
      approved = Math.floor((approved / len) * 100)
      pending = Math.floor((pending / len) * 100)
      rejected = Math.floor((rejected / len) * 100)
      rest = 100 - approved - pending - rejected
    } else {
      rest = 100
    }

    return (
      <div className="partnerBox">
        <div className="duedate">
          <div className="due">Due</div>
          {partner.duedate}
        </div>
        <div className="iconBox">{partner.org_name[0]}</div>
        <div className="nameProgressDisplay">
          <div>{partner.org_name}</div>
          <div className="progressAdditional">
            {approved}%
            <Progress multi>
              <Progress bar color="dashgreen" value={approved} />
              <Progress bar color="dashorange" value={pending} />
              <Progress bar color="dashred" value={rejected} />
              <Progress bar color="dashgrey" value={rest} />
            </Progress>
          </div>
        </div>
      </div>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PMMainPage)
