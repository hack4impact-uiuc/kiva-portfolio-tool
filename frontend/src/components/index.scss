body, html {
    height: 100%;
    width: 100%;
    padding: 0;
    border: 0 none;
    margin: 0;
}

.container {
    height: 100%;
    width: 100%;
}