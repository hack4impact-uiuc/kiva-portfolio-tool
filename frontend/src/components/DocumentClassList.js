import React, { Component } from 'react'
import { Table } from 'reactstrap'
import DocumentClass from './DocumentClass'
import '../styles/documentlist.css'

class DocumentClassList extends Component {
  render() {
    return (
      <Table>
        <tbody>
          <tr>
            <th className="theader-centered">Document Class Name</th>
            <th />
          </tr>
          {this.props.documentClasses
            ? this.props.documentClassess.map(documentClass => (
                <DocumentClass documentClass={documentClass} />
              ))
            : null}
        </tbody>
      </Table>
    )
  }
}

export default DocumentClassList
