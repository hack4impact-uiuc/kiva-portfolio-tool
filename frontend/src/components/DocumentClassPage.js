import React, { Component } from 'react'
import { Button, Modal, ModalBody, ModalFooter, Table, Input } from 'reactstrap'
import Dropzone from 'react-dropzone'

import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { updateDocumentClasses, beginLoading, endLoading } from '../redux/modules/user'

import WithAuth from './auth/WithAuth'
import DocumentClass from './DocumentClass'
import NavBar from './NavBar'

import { getAllDocumentClasses, createDocumentClass } from '../utils/ApiWrapper'

import add from '../media/add.png'

import '../styles/variables.scss'
import '../styles/documentclasspage.scss'

const mapStateToProps = state => ({
  documentClasses: state.user.documentClasses,
  language: state.user.language
})

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      updateDocumentClasses,
      beginLoading,
      endLoading
    },
    dispatch
  )
}

/*
A page accessible by admins and PMs with an overview of all Document Classes
Functionality: Allows creation of new Document Classes and viewing/editing/deletion of existing ones through the DocumentClass component
*/
export class DocumentClassPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      addModal: false, // modal that appears after clicking 'Add New Document Class'
      name: '',
      description: '',
      files: []
    }
    this.toggle = this.toggle.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  /**
   * Gets all document classes once component is ready
   */
  async componentDidMount() {
    const document_classes = await getAllDocumentClasses()
    if (document_classes) {
      this.props.updateDocumentClasses(document_classes)
    } else {
      this.props.updateDocumentClasses([])
    }
  }

  /**
   * Opens/Closes the addModal
   */
  toggle() {
    this.setState({ addModal: !this.state.addModal })
  }

  /**
   * updates name in state
   */
  updateName = event => {
    this.setState({ name: event.target.value })
  }

  /**
   * updates description in state
   */
  updateDescription = event => {
    this.setState({ description: event.target.value })
  }

  /**
   * updates document class and closes add modal
   */
  async handleSubmit() {
    this.props.beginLoading()
    await createDocumentClass(
      this.state.name,
      this.state.description,
      this.state.files[0],
      this.state.files[0].name
    )
    const document_classes = await getAllDocumentClasses()
    if (document_classes) {
      this.props.updateDocumentClasses(document_classes)
    } else {
      this.props.updateDocumentClasses([])
    }
    this.props.endLoading()
    this.toggle()
  }

  onDrop(files) {
    this.setState({
      files
    })
  }

  languages = {
    English: {
      name: 'Name:',
      description: 'Description:',
      fileUploaded: 'File uploaded: ',
      click: 'Click to upload',
      return: 'Return',
      close: 'Close',
      create: 'Create new document class',
      edit: 'Edit document classes',
      add: 'Add new document class',
      docClassName: 'Document class name'
    },
    Spanish: {
      name: 'Name: (Spanish)',
      description: 'Description: (Spanish)',
      fileUploaded: 'File uploaded:  (Spanish)',
      click: 'Click to upload (Spanish)',
      return: 'Return (Spanish)',
      close: 'Close (Spanish)',
      create: 'Create new document class (Spanish)',
      edit: 'Edit document classes (Spanish)',
      add: 'Add new document class (Spanish)',
      docClassName: 'Document class name (Spanish)'
    },
    French: {
      name: 'Name: (French)',
      description: 'Description: (French)',
      fileUploaded: 'File uploaded:  (French)',
      click: 'Click to upload (French)',
      return: 'Return (French)',
      close: 'Close (French)',
      create: 'Create new document class (French)',
      edit: 'Edit document classes (French)',
      add: 'Add new document class (French)',
      docClassName: 'Document class name (French)'
    },
    Portuguese: {
      name: 'Name: (Portuguese)',
      description: 'Description: (Portuguese)',
      fileUploaded: 'File uploaded:  (Portuguese)',
      click: 'Click to upload (Portuguese)',
      return: 'Return (Portuguese)',
      close: 'Close (Portuguese)',
      create: 'Create new document class (Portuguese)',
      edit: 'Edit document classes (Portuguese)',
      add: 'Add new document class (Portuguese)',
      docClassName: 'Document class name (Portuguese)'
    }
  }

  render() {
    let text = this.languages[this.props.language]
    if (!text) {
      text = this.languages['English']
    }

    return (
      <div className="background-rectangles maxheight">
        <NavBar />
        <Modal isOpen={this.state.addModal} toggle={this.toggle}>
          <ModalBody>
            <form>
              <span>{text.name}</span>
              <Input type="textarea" className="textarea-input" onChange={this.updateName} />
              <br />
              <span>{text.description}</span>
              <Input
                type="textarea"
                className="textarea-input"
                style={{ height: '200px' }}
                onChange={this.updateDescription}
              />
              <br />
              <div className="dropPage">
                <section className="droppedBox">
                  <div className="dropZone">
                    <Dropzone onDrop={this.onDrop.bind(this)}>
                      {({ getRootProps, getInputProps }) => (
                        <section>
                          <div {...getRootProps()}>
                            <input {...getInputProps()} />
                            {this.state.files.length > 0 ? (
                              <p>{text.fileUploaded + this.state.files[0].name}</p>
                            ) : (
                              <p>{text.click}</p>
                            )}
                          </div>
                        </section>
                      )}
                    </Dropzone>
                  </div>
                </section>
                <hr />
              </div>
            </form>
          </ModalBody>
          <ModalFooter>
            <Button className="invalidSearchButton" onClick={this.toggle}>
              {text.close}
            </Button>
            <Button
              disabled={this.state.files.length === 0}
              color="success"
              onClick={this.handleSubmit}
            >
              {text.create}
            </Button>
          </ModalFooter>
        </Modal>
        <div className="edit-banner">
          <h1 className="h1">{text.edit}</h1>
          <Button
            className="add-doc-text"
            id="add-new-class"
            color="transparent"
            onClick={this.toggle}
          >
            <img className="addImg" src={add} alt="Add icon" />
            <span className="add-doc-text">{text.add}</span>
          </Button>
        </div>

        <div className="doc-table">
          <Table>
            <tbody>
              <tr id="doc-table-top-row">
                <th className="theader-centered">{text.docClassName}</th>
                <th />
              </tr>
              {this.props.documentClasses
                ? this.props.documentClasses.map(documentClass => (
                    <DocumentClass documentClass={documentClass} />
                  ))
                : null}
            </tbody>
          </Table>
        </div>

        <div className="returnBtnContainer text-centered padding-bottom-sm">
          <Button className="returnButton" onClick={this.props.history.goBack}>
            {text.return}
          </Button>
        </div>
      </div>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WithAuth(DocumentClassPage))
