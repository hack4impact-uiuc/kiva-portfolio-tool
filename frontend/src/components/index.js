import Dashboard from './Dashboard'
import DocumentPreview from './DocumentPreview'

import SelectDocumentsPage from './SelectDocuments'
import BoxPreview from './DocumentPreview'
import PMMainPage from './PMMainPage'
import DocumentClassPage from './DocumentClassPage.js'

import Load from './Load'

import ErrorMessage from './auth/ErrorMessage'
import ForgotPassword from './auth/ForgotPassword'
import LogIn from './auth/LogIn'
import Register from './auth/Register'
import WrongPage from './auth/WrongPage'

export {
  Dashboard,
  LogIn,
  BoxPreview,
  Register,
  SelectDocumentsPage,
  PMMainPage,
  ForgotPassword,
  DocumentClassPage,
  DocumentPreview,
  Load,
  WrongPage,
  ErrorMessage
}
