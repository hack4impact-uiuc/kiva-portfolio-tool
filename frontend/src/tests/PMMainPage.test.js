import React from 'react'
import { PMMainPage } from '../components/PMMainPage'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson, mountToJson } from 'enzyme-to-json'
import { configure, shallow, mount } from 'enzyme'
import chai from 'chai'
global.jestExpect = global.expect
global.expect = chai.expect
import sinon from 'sinon'

beforeAll(() => {
  configure({ adapter: new Adapter() })
})

test('PMMainPage renders correctly', () => {
  const component = mount(<PMMainPage />)

  let tree = mountToJson(component)
  jestExpect(tree).toMatchSnapshot()
})

describe('<PMMainPage />', () => {
  it('calls componentDidMount', () => {
    sinon.spy(PMMainPage.prototype, 'componentDidMount')
    const wrapper = mount(<PMMainPage />)
    expect(PMMainPage.prototype.componentDidMount).to.have.property('callCount', 1)
    PMMainPage.prototype.componentDidMount.restore()
  })
})
