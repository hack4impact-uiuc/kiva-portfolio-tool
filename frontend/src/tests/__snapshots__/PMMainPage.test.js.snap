// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PMMainPage renders correctly 1`] = `
<PMMainPage>
  <div
    className="page"
  >
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropTransition={
        Object {
          "mountOnEnter": true,
          "timeout": 150,
        }
      }
      centered={false}
      fade={true}
      isOpen={false}
      keyboard={true}
      modalTransition={
        Object {
          "timeout": 300,
        }
      }
      onClosed={[Function]}
      onOpened={[Function]}
      role="dialog"
      toggle={[Function]}
      zIndex={1050}
    />
    <div>
      <span />
      <p>
        Fieldy McPartnerson
      </p>
    </div>
    <h2>
      Field Partners
    </h2>
    <Button
      color="secondary"
      onClick={[Function]}
      tag="button"
    >
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Add New Field Partner
      </button>
    </Button>
    <form>
      <img
        src="search.png"
        width="25"
      />
      <input
        className="input-master"
        onChange={[Function]}
        placeholder="Search for a Field Partner..."
        type="text"
        value=""
      />
    </form>
    <Tabs
      className="tab-master"
      defaultFocus={false}
      defaultIndex={null}
      forceRenderTabPanel={false}
      selectedIndex={null}
    >
      <UncontrolledTabs
        className="tab-master"
        focus={false}
        forceRenderTabPanel={false}
        onSelect={[Function]}
        selectedIndex={0}
      >
        <div
          className="tab-master"
          data-tabs={true}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <TabList
            className="react-tabs__tab-list"
            key=".0"
          >
            <ul
              className="react-tabs__tab-list"
              role="tablist"
            >
              <Tab
                className="react-tabs__tab"
                disabledClassName="react-tabs__tab--disabled"
                focus={false}
                id="react-tabs-0"
                key=".0"
                panelId="react-tabs-1"
                selected={true}
                selectedClassName="react-tabs__tab--selected"
                tabRef={[Function]}
              >
                <li
                  aria-controls="react-tabs-1"
                  aria-disabled="false"
                  aria-selected="true"
                  className="react-tabs__tab react-tabs__tab--selected"
                  id="react-tabs-0"
                  role="tab"
                  tabIndex="0"
                >
                  In Process
                </li>
              </Tab>
              <Tab
                className="react-tabs__tab"
                disabledClassName="react-tabs__tab--disabled"
                focus={false}
                id="react-tabs-2"
                key=".1"
                panelId="react-tabs-3"
                selected={false}
                selectedClassName="react-tabs__tab--selected"
                tabRef={[Function]}
              >
                <li
                  aria-controls="react-tabs-3"
                  aria-disabled="false"
                  aria-selected="false"
                  className="react-tabs__tab"
                  id="react-tabs-2"
                  role="tab"
                  tabIndex={null}
                >
                  New Partner
                </li>
              </Tab>
              <Tab
                className="react-tabs__tab"
                disabledClassName="react-tabs__tab--disabled"
                focus={false}
                id="react-tabs-4"
                key=".2"
                panelId="react-tabs-5"
                selected={false}
                selectedClassName="react-tabs__tab--selected"
                tabRef={[Function]}
              >
                <li
                  aria-controls="react-tabs-5"
                  aria-disabled="false"
                  aria-selected="false"
                  className="react-tabs__tab"
                  id="react-tabs-4"
                  role="tab"
                  tabIndex={null}
                >
                  Complete
                </li>
              </Tab>
            </ul>
          </TabList>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            id="react-tabs-1"
            key=".1"
            selected={true}
            selectedClassName="react-tabs__tab-panel--selected"
            tabId="react-tabs-0"
          >
            <div
              aria-labelledby="react-tabs-0"
              className="react-tabs__tab-panel react-tabs__tab-panel--selected"
              id="react-tabs-1"
              role="tabpanel"
            >
              <div
                className="partnerPanel"
              />
            </div>
          </TabPanel>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            id="react-tabs-3"
            key=".2"
            selected={false}
            selectedClassName="react-tabs__tab-panel--selected"
            tabId="react-tabs-2"
          >
            <div
              aria-labelledby="react-tabs-2"
              className="react-tabs__tab-panel"
              id="react-tabs-3"
              role="tabpanel"
            />
          </TabPanel>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            id="react-tabs-5"
            key=".3"
            selected={false}
            selectedClassName="react-tabs__tab-panel--selected"
            tabId="react-tabs-4"
          >
            <div
              aria-labelledby="react-tabs-4"
              className="react-tabs__tab-panel"
              id="react-tabs-5"
              role="tabpanel"
            />
          </TabPanel>
        </div>
      </UncontrolledTabs>
    </Tabs>
  </div>
</PMMainPage>
`;
